### PR TITLE
Clean up worktree dirs during sync loop

### PR DESCRIPTION
**Pull request by thockin**
Avoid "already exists" error.  This is a port of #412 but I changed
the e2e to use `docker pause`.

### CODE REVIEW

File: test_e2e.sh [hunk 1]
```diff
@@ -424,6 +424,54 @@ assert_file_eq "$ROOT"/link/file "$TESTCASE 1"
...
 
+##############################################
+# Test worktree-cleanup
+##############################################
+testcase "worktree-cleanup"
+
+echo "$TESTCASE" > "$REPO"/file
+git -C "$REPO" commit -qam "$TESTCASE"
+GIT_SYNC \
+    --period=100ms \
```
Comment (barney-s) [inline test_e2e.sh:435]: This needs to be --wait

**Comment by barney-s** [comment]
@thockin 
Sorry pulled the trigger too early.
The parameter "--period" needs to be "--wait" ?

**Comment by thockin** [comment]
On v4 it is --period

**Pull request merged**